packages/server/src/api/routes/public/tests/users.spec.ts

```diff
@@ -1,132 +1,121 @@
 import * as setup from "../../tests/utilities"
-import { generateMakeRequest, MakeRequestResponse } from "./utils"
 import { User } from "@budibase/types"
-import { mocks } from "@budibase/backend-core/tests"
+import { generator, mocks } from "@budibase/backend-core/tests"
+import nock from "nock"
+import environment from "../../../../environment"
+import TestConfiguration from "../../../../tests/utilities/TestConfiguration"
 
-import * as workerRequests from "../../../../utilities/workerRequests"
-
-const mockedWorkerReq = jest.mocked(workerRequests)
-
-let config = setup.getConfig()
-let apiKey: string, globalUser: User, makeRequest: MakeRequestResponse
+const config = new TestConfiguration()
+let globalUser: User
+let users: Record<string, User> = {}
 
 beforeAll(async () => {
   await config.init()
-  globalUser = await config.globalUser()
-  apiKey = await config.generateApiKey(globalUser._id)
-  makeRequest = generateMakeRequest(apiKey)
-  mockedWorkerReq.readGlobalUser.mockImplementation(() =>
-    Promise.resolve(globalUser)
-  )
 })
 
 afterAll(setup.afterAll)
 
-function base() {
-  return {
-    tenantId: config.getTenantId(),
-    firstName: "Test",
-    lastName: "Test",
-  }
-}
-
-function updateMock() {
-  mockedWorkerReq.readGlobalUser.mockImplementation(ctx => ctx.request.body)
-}
+beforeEach(async () => {
+  globalUser = await config.globalUser()
+  users[globalUser._id!] = globalUser
+
+  nock.cleanAll()
+  nock(environment.WORKER_URL!)
+    .get(new RegExp(`/api/global/users/.*`))
+    .reply(200, (uri, body) => {
+      const id = uri.split("/").pop()
+      return users[id!]
+    })
+    .persist()
+
+  nock(environment.WORKER_URL!)
+    .post(`/api/global/users`)
+    .reply(200, (uri, body) => {
+      const newUser = body as User
+      if (!newUser._id) {
+        newUser._id = `us_${generator.guid()}`
+      }
+      users[newUser._id!] = newUser
+      return newUser
+    })
+    .persist()
```

Comment:
These two HTTP mocks create a very crude implementation of the worker API for creating and retrieving users. It just takes what it's given and stores it into a map by the user ID.

In an ideal world, I'd like to _actually_ hit the worker API here. I don't see why we wouldn't be able to. That or we start the work to merge the server and worker packages.